### PR TITLE
Qmaps 153

### PR DESCRIPTION
**Pull request by Julien-laville**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "@qwant/map-style-builder",
-  "version": "0.1.0",
+  "version": "0.2.0",
   "description": "Some tools to build and debug GL Style at Qwant",
   "main": "./task/run.js",
   "scripts": {
     "test": "node task/test.js --style_dir ",
-    "build_all": "bash task/build.sh "
+    "build_all": "node task/build_cli.js"
```
Comment (nlehuby) [inline package.json:8]: You need to update the readme.
`npm run build_all -- PATH/TO/YOUR/MAPSTYLE/FOLDER PATH/TO/YOUR/CONF` does not work anymore
`npm run build_all -- --style_dir PATH/TO/YOUR/MAPSTYLE/FOLDER --conf PATH/TO/YOUR/CONF` is ok

File: lib/build.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        return {
+          svg: fs.readFileSync(f),
+          id: path.basename(f).replace('.svg', '')
+        }
+      })
+
+      options.pixelRatios = options.pixelRatios || [1]
+      options.pixelRatios.forEach((pixelRatio) => {
+      let pixelRatioPath = `sprite${pixelRatio > 1 ? `@${pixelRatio}x` : ''}`
+      let jsonPath = `${options.outPath}${pixelRatioPath}.json`
```
Comment (nlehuby) [inline lib/build.js]: The sprite json and png files are not generated in the outPath (missing a `/`)

File: task/build_cli.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const jsonconf = JSON.parse(confStr);
+
+let options = {
+  styleDir : args.style_dir,
+  conf : jsonconf,
+  outPath : buildDir
+}
+mkdirp(buildDir)
+const builtStyle = build(styleStr, options)
+outPath = path.join(buildDir,'built-style.json');
```
Comment (antoine-de) [inline task/build_cli.js]: you need to change the options too

File: task/deploy.sh [hunk 4]
```diff
@@ -1,25 +0,0 @@
...
-  npm install spritezero-cli
-fi
-
-rm -rf build
-mkdir build
-if [ -d ../icons ]; then
-  ./node_modules/.bin/spritezero build/sprite ../icons/
-  ./node_modules/.bin/spritezero --retina build/sprite@2x ../icons/
-fi
-node task/deploy.js
```
Comment (nlehuby) [inline task/deploy.sh:15]: You can remove the deploy.js file too. We don't use it.

File: lib/build.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  const tileSchemaPoiPath = path.join(styleDir,'tileschema_poi.json')
+  const tileSchemaBaseStr = fs.readFileSync(tileSchemaBasePath, 'utf8')
+  const tileSchemaPoiStr = fs.readFileSync(tileSchemaPoiPath, 'utf8')
+
+  const tileSchemaBase = JSON.parse(tileSchemaBaseStr)
+  const tileSchemaPoi = JSON.parse(tileSchemaPoiStr)
+
+  if(options.ouput === 'dev') {
+    transform.adjustStyleWithoutTilejson({
+      style: style,
```
Comment (nlehuby) [inline lib/build.js]: You need to parse the style as JSON before trying to transform it.

File: lib/build.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            console.error(err)
+            return
+          }
+          fs.writeFileSync(pngPath, image)
+        })
+      })
+    })
+
+  }
+  return JSON.stringify(style)
```
Comment (nlehuby) [inline lib/build.js:79]: You already stringify it in the `build_cli.js` file.

File: lib/build.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  const needSprite = fs.existsSync(path.join(styleDir,'icons'))
+  const tileSchemaBasePath = path.join(styleDir,'tileschema_base.json')
+  const tileSchemaPoiPath = path.join(styleDir,'tileschema_poi.json')
+  const tileSchemaBaseStr = fs.readFileSync(tileSchemaBasePath, 'utf8')
+  const tileSchemaPoiStr = fs.readFileSync(tileSchemaPoiPath, 'utf8')
+
+  const tileSchemaBase = JSON.parse(tileSchemaBaseStr)
+  const tileSchemaPoi = JSON.parse(tileSchemaPoiStr)
+
+  if(options.ouput === 'dev') {
```
Comment (nlehuby) [inline lib/build.js]: `options.output` is not set in `build_cli.js`

**Review by nlehuby — CHANGES_REQUESTED** [review]

File: lib/build.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  const needSprite = fs.existsSync(path.join(styleDir,'icons'))
+  const tileSchemaBasePath = path.join(styleDir,'tileschema_base.json')
+  const tileSchemaPoiPath = path.join(styleDir,'tileschema_poi.json')
+  const tileSchemaBaseStr = fs.readFileSync(tileSchemaBasePath, 'utf8')
+  const tileSchemaPoiStr = fs.readFileSync(tileSchemaPoiPath, 'utf8')
+
+  const tileSchemaBase = JSON.parse(tileSchemaBaseStr)
+  const tileSchemaPoi = JSON.parse(tileSchemaPoiStr)
+
+  if(options.ouput === 'production') {
```
Comment (nlehuby) [inline lib/build.js]: You need to change `ouput` to `output` ;)

File: task/build_cli.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+const buildDir = path.join(args.style_dir, 'build')
+const stylePath = path.join(args.style_dir,'style.json')
+const confStr = fs.readFileSync(args.conf, 'utf8')
+const styleStr = fs.readFileSync(stylePath, 'utf8')
+const style = JSON.parse(styleStr)
+const jsonconf = JSON.parse(confStr)
+
+let options = {
+  styleDir : args.style_dir,
+  conf : jsonconf,
```
Comment (nlehuby) [inline task/build_cli.js]: `options.conf` is defined here, whereas `options.config` is expected in `build_cli`

**Review by nlehuby — APPROVED** [review]

**Pull request merged**